tests/test_aws.py

```diff
@@ -1311,6 +1323,117 @@ def test_retrieve_subject_token_success_temp_creds_no_environment_vars_idmsv2(
             },
         )
 
+    def test_validate_metadata_server_url_if_any(self):
```

Comment:
you can probably merge most of these tests in a for loop, would be cleaner IMO. up to you